### DISCUSSION TITLE
Do I need to install ImageMagick to use C++ API?

### DISCUSSION THREAD

**Discussion by hsestagio** [ImageMagick/ImageMagick#4265] [Help]
I have a computer that doesn't have ImageMagick installed. I tried to copy all DLLs to that computer, but it wouldn't work. Has anyone suceeded using ImageMagick C++ APIs without installing the ImageMagick?

Thanks

**Comment by dlemstra** [comment]
You could decide to statically link ImageMagick in your application and then you don't need to install ImageMagick on the computer. Since you are using the term DLL I assume that you are working on Windows. It might also be possible that you will need to install the C++ redistributable packages on that machine.

**Reply by hsestagio** [reply]
It didnt work, gives me this error "Magick: RegistryKeyLookupFailed `CoderModulesPath' @ error/module.c/GetMagickModulePath/665File: C:\Progr..."

**Reply by hsestagio** [reply]
I added coder to env. variables and now im getting this error "Magick: FailedToExecuteCommand `"gs" -sstdout=%stderr -dQUIET -dSAFER -dBATCH -dNOPAUSE -dNOPROMPT -dMaxBitmap=500000000 -dAlignToPixels=0 -dGridFitTT=2 "-sDEVICE=pngalpha" -dTextAlphaBits=4 -dGraphicsAlphaBits=4 "-r300x300" -g2479x3508  "-sOutputFile=C:/Users/User/AppData/Local/Temp/magick-BlejNdpOb3_JRQteTWVwcx35670_HH9x%d" "-fC:/Users/User/AppData/Local/Temp/magick-nUYhi_Ep7PsM2qM6UJybG0KMN4hqpOFF" "-fC:/Users/User/AppData/Local/Temp/magick-uyz5Si1eB0dUV2GXBQUR26bf7SCqxz88"' (O sistema n?o conseguiu localizar o ficheiro especificado."

**Reply by snibgo** [reply]
Is Ghostscript installed? If not, that is the problem.

**Reply by hsestagio** [reply]
It is, magick++ works well while imagemagick is installed, after uninstalling imagemagick it throws this errors.
